src/client/driver/index.js

```diff
@@ -0,0 +1,105 @@
+import hammerhead from './deps/hammerhead';
+import testCafeCore from './deps/testcafe-core';
+import testCafeUI from './deps/testcafe-ui';
+
+import MESSAGE from '../../test-run/client-messages';
+import COMMAND_TYPE from '../../test-run/commands/type';
+import { UncaughtJSError } from '../../errors/test-run';
+
+import * as browser from '../browser';
+import executeActionCommand from './command-executors/execute-action-command';
+import ContextStorage from './storage';
+
+var transport         = hammerhead.transport;
+var XhrBarrier        = testCafeCore.XhrBarrier;
+var pageUnloadBarrier = testCafeCore.pageUnloadBarrier;
+var eventUtils        = testCafeCore.eventUtils;
+var modalBackground   = testCafeUI.modalBackground;
+
+
+const executingCommandFlag = 'testcafe|driver|executing-command-flag';
+
+
+export default class ClientDriver {
+    constructor (testRunId, heartbeatUrl, browserStatusUrl) {
+        this.testRunId        = testRunId;
+        this.heartbeatUrl     = heartbeatUrl;
+        this.browserStatusUrl = browserStatusUrl;
+        this.contextStorage   = new ContextStorage(window, testRunId);
+
+        this.pageInitialXhrBarrier = new XhrBarrier();
+
+        pageUnloadBarrier.init();
+    }
+
+    start () {
+        browser.startHeartbeat(this.heartbeatUrl, hammerhead.nativeMethods.XMLHttpRequest);
```

Comment:
Yes, I tried to do this but there is some difficult with this. It should be more complex implementation than I expected. I prefer to do this in the context of the new issue, because this PR is huge already